From 98b7ca9279e9b4b2499b4c796b64a99035d25a0a Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Wed, 20 Aug 2025 15:36:03 +0200
Subject: [PATCH] winnt: define __MINGW_CXX1[1|4]_CONSTEXPR found in _mingw.h

When installing winnt.h in contribs this is not defined with mingw-w64
older than v13.0.0.
---
 mingw-w64-headers/include/winnt.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/mingw-w64-headers/include/winnt.h b/mingw-w64-headers/include/winnt.h
index 38b4d4f38..296a9ec3c 100644
--- a/mingw-w64-headers/include/winnt.h
+++ b/mingw-w64-headers/include/winnt.h
@@ -710,6 +710,19 @@ typedef LONG RTL_REFERENCE_COUNT32, *PRTL_REFERENCE_COUNT32;
 #define RTL_CONST_CAST(type) (type)
 #endif
 
+#define __MINGW_CXX11_CONSTEXPR
+#define __MINGW_CXX14_CONSTEXPR
+#ifdef __cplusplus
+# if __cplusplus >= 201103L || defined(_MSC_VER)
+#  undef __MINGW_CXX11_CONSTEXPR
+#  define __MINGW_CXX11_CONSTEXPR  constexpr
+# endif
+# if __cplusplus >= 201402L || defined(_MSC_VER)
+#  undef __MINGW_CXX14_CONSTEXPR
+#  define __MINGW_CXX14_CONSTEXPR  constexpr
+# endif
+#endif
+
 #ifdef __cplusplus
 #define DEFINE_ENUM_FLAG_OPERATORS(ENUMTYPE) \
 extern "C++" { \
-- 
2.45.1.windows.1

